crates/bevy_render/src/texture/image_texture_conversion.rs

```diff
@@ -1,194 +1,229 @@
+                data = cast_slice(&raw_data).to_owned();
+            }
+            DynamicImage::ImageRgb32F(image) => {
+                width = image.width();
+                height = image.height();
+                format = TextureFormat::Rgba32Float;
+
+                let mut local_data =
+                    Vec::with_capacity(width as usize * height as usize * format.pixel_size());
+
+                for pixel in image.into_raw().chunks_exact(3) {
+                    // TODO: use the array_chunks method once stabilised
+                    // https://github.com/rust-lang/rust/issues/74985
+                    let r = pixel[0];
+                    let g = pixel[1];
+                    let b = pixel[2];
+                    let a = u16::max_value();
+
+                    local_data.extend_from_slice(&r.to_ne_bytes());
+                    local_data.extend_from_slice(&g.to_ne_bytes());
+                    local_data.extend_from_slice(&b.to_ne_bytes());
+                    local_data.extend_from_slice(&a.to_ne_bytes());
+                }
+
+                data = local_data;
             }
+            DynamicImage::ImageRgba32F(image) => {
+                width = image.width();
+                height = image.height();
+                format = TextureFormat::Rgba32Float;
 
-            data = local_data;
+                let raw_data = image.into_raw();
+
+                data = cast_slice(&raw_data).to_owned();
+            }
+            // DynamicImage is now non exhaustive, catch future variants and convert them
+            _ => {
+                let image = dyn_img.into_rgba8();
+                width = image.width();
+                height = image.height();
+                format = TextureFormat::Rgba8UnormSrgb;
+
+                data = image.into_raw();
+            }
         }
-        DynamicImage::ImageRgba32F(image) => {
-            width = image.width();
-            height = image.height();
-            format = TextureFormat::Rgba32Float;
 
-            let raw_data = image.into_raw();
+        Image::new(
+            Extent3d {
+                width,
+                height,
+                depth_or_array_layers: 1,
+            },
+            TextureDimension::D2,
+            data,
+            format,
+        )
+    }
 
-            data = cast_slice(&raw_data).to_owned();
-        }
-        // DynamicImage is now non exhaustive, catch future variants and convert them
-        _ => {
-            let image = dyn_img.into_rgba8();
-            width = image.width();
-            height = image.height();
-            format = TextureFormat::Rgba8UnormSrgb;
-
-            data = image.into_raw();
+    /// Converts an [`Image`] to a [`DynamicImage`]. Usefull for editing the
+    /// contents of an Image. Not all [`TextureFormat`] are covered, therefore
+    /// it will return `None` if the format is unsupported. The supported
+    /// formats are the same as those listed for `Image::convert`.
```

Comment:
```suggestion
    /// formats are the same as those listed for [`Image::convert`].
```

I would provide a link here so that the user can directly click it to refer to the `Image::convert` doc.

Nit: I'd also list the supported formats in this method and provide a link to this method in `Image::convert` instead, but that's non-blocking.

Reply:
I forgot to change that the fact that method now returns a `Result` instead of an `Option` the first time around so I changed that too. Hopefully its clear now.